weave-js/src/components/PagePanelComponents/Home/Browse3/context.tsx

```diff
@@ -664,6 +665,27 @@ export const usePeekLocation = () => {
   }, [peekPath]);
 };
 
+export const useClosePeekOrNavigateUp = () => {
+  /*
+  If we are peeking, just close the peek drawer
+  If we are not peeking, navigate up one level, unless we are at the root
+  */
+  const history = useHistory();
+  const peekLoc = usePeekLocation();
+  const closePeek = useClosePeek();
+
+  return () => {
+    if (peekLoc != null) {
+      closePeek();
+    } else {
+      // Not peeking (fullscreen) so navigate to the parent page
+      const path = history.location.pathname.split('/');
```

Comment:
i'm not quite sure what the intention is here

Reply:
This is totally a hack, but when you are in fullscreen mode (which you can navigate to directly with a link) there is nothing to close once you have deleted the call you are looking at. So we have to navigate up the tree after deletion as the current page is no longer valid...